libs/platform/user/ebpf_platform_user.cpp

```diff
@@ -57,11 +57,22 @@ static std::vector<uint32_t> _ebpf_platform_group_to_index_map;
 static ebpf_result_t
 _initialize_thread_pool()
 {
+    memset(&_callback_environment, 0, sizeof(_callback_environment));
+
+    // CreateThreadpoolCleanupGroup can return nullptr.
+    if (ebpf_fault_injection_is_enabled() && ebpf_fault_injection_inject_fault()) {
```

Comment:
Ok, but it would be cleaner to not have 2 functions here but make ebpf_fault_injection_inject_fault() return false if !ebpf_fault_injection_is_enabled()

Reply:
Did you mean,
if (ebpf_fault_injection_is_enabled()) {
       if (ebpf_fault_injection_inject_fault()) {       // if true
           return EBPF_NO_MEMORY;
       }
}

If it is false, the call flow will proceed as normal.

Reply:
Removed ebpf_fault_injection_is_enabled()